docs/src/reference/experimental/bounded_arbitrary.md

```diff
@@ -0,0 +1,80 @@
+# Bounded Non-deterministic variables
+
+This is an experimental feature that allows you to bound otherwise unbounded types. For example, `Vec<T>` does not have an `Arbitrary` implementation because vectors can grow arbitrarily in size. One way of handling proofs about such types is to make the problem easier, and only prove a property up to some bound. Of course, the proof is only valid up to the bound, but can still be useful in providing confidence that your code is correct.
+
+## Example
+
+As a toy example, let's prove, up to some bound, that reversing a vector twice gives you back the original vector. Here's a reversing function:
+
+```rust
+fn reverse_vector<T>(mut input: Vec<T>) -> Vec<T> {
+    let mut reversed = vec![];
+    for _ in 0..input.len() {
+        reversed.push(input.pop().unwrap());
+    }
+    reversed
+}
+```
+
+We can use `BoundedAny` to write a proof harness:
+
+```rust
+#[kani::proof]
+fn check_reverse_is_its_own_inverse() {
+    // We use BoundedAny to construct a vector that has at most length 16
+    let input: BoundedAny<Vec<bool>, 16> = kani::any();
+
+    let double_reversed = reverse_vector(reverse_vector(input.clone().into_inner()));
+
+    // we assert that every value in the input is the same as the value in the
+    // doubly reversed list
+    for i in 0..input.len() {
+        assert!(input[i] == double_reversed[i])
+    }
+}
+```
+
+Then, with `kani --harness check --unwind 17` we can prove that, for vectors up to size 16, our reverse function is indeed it's own inverse.
```

Comment:
You can add `#[kani::unwind(17)]` to avoid needing the command line. You can also omit the `--harness`. Probably don't need to mention the command at all.